### PR TITLE
Adds experimental support for LangChain callbacks to be passed to mlflow.langchain models via a protected method.

### PR DESCRIPTION
**Pull request by AveshCSingh**
### What changes are proposed in this pull request?

Introduces a protected method _predict_with_callbacks to add support to pass LangChain callbacks to a model. 

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [x] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by AveshCSingh** [comment]
> In addition to the offline discussion items, we'll want to make sure that we're validating this feature functions properly within our model serving test fixture and that there are no impacts to signature enforcement / validation.

Sounds good. I'll add a test that includes pyfunc_serve_and_score_model.

File: mlflow/langchain/__init__.py [hunk 1]
```diff
@@ -460,16 +460,34 @@ def predict(  # pylint: disable=unused-argument
         data: Union[pd.DataFrame, List[Union[str, Dict[str, Any]]], Any],
         params: Optional[Dict[str, Any]] = None,  # pylint: disable=unused-argument
     ) -> List[str]:
-        """
-        :param data: Model input data.
-        :param params: Additional parameters to pass to the model for inference.
+        messages = self._prepare_messages(data)
```
Comment (BenWilson2) [inline mlflow/langchain/__init__.py:463]: Could we bring the pydoc string back?

File: mlflow/langchain/__init__.py [hunk 2]
```diff
@@ -460,16 +460,34 @@ def predict(  # pylint: disable=unused-argument
...
 
-        :return: Model predictions.
-        """
+    @experimental
+    def _predict_with_callbacks(  # pylint: disable=unused-argument
+        self,
+        data: Union[pd.DataFrame, List[Union[str, Dict[str, Any]]], Any],
+        params: Optional[Dict[str, Any]] = None,  # pylint: disable=unused-argument
+        callback_handlers=None,
+    ) -> List[str]:
```
Comment (BenWilson2) [inline mlflow/langchain/__init__.py:478]: Can we add a brief pydoc string here just for future reference purposes to explain why we have this private method here?

**Review by BenWilson2 — APPROVED** [review]
2 tiny nits; otherwise LGTM!

File: mlflow/langchain/__init__.py (hunk 2 above)
Comment (AveshCSingh) [reply mlflow/langchain/__init__.py:478]: I accidentally removed these during the refactor. Added back.

**Review by dbczumar — APPROVED** [review]
LGTM!

**Pull request merged**